### PR TITLE
Multi send - Tonkeeper Pro

### PR DESCRIPTION
**Pull request by siandreev**

### CODE REVIEW

File: packages/core/src/service/transfer/jettonService.ts [hunk 1]
```diff
@@ -18,9 +19,10 @@ import {
     getWalletBalance,
     SendMode
 } from './common';
+import { TransferMessage, transferMessagesToGroups } from './tonService';
 
-const jettonTransferAmount = toNano('0.64');
-const jettonTransferForwardAmount = BigInt('1');
+const jettonTransferAmount = toNano(0.1);
+const jettonTransferForwardAmount = toNano(0.01);
```
Comment (KuznetsovNikita) [inline packages/core/src/service/transfer/jettonService.ts]: Previously, it was set at 1 nano TON; at such a small amount, twm are not launch and transactions were not process and errors could not occur.
Please return 1 nano TON

File: packages/core/src/service/transfer/tonService.ts [hunk 2]
```diff
@@ -275,3 +276,141 @@ export const sendTonTransfer = async (
...
+                })
+            )
+        };
+        const transfer = contract.createTransfer(raw);
+        return { packed: externalMessage(contract, seqno, transfer).toBoc(), raw };
+    });
+};
+
+export const MAX_ALLOWED_MULTI_TRANSFERS = {
+    [WalletVersion.W5]: 510,
```
Comment (KuznetsovNikita) [inline packages/core/src/service/transfer/tonService.ts]: Are you sure that - 510 internal messages?

File: packages/core/src/service/transfer/tonService.ts [hunk 3]
```diff
@@ -275,3 +276,141 @@ export const sendTonTransfer = async (
         sendBlockchainMessageRequest: { boc: cell.toString('base64') }
     });
 };
+
+export type TransferMessage = {
```
Comment (KuznetsovNikita) [inline packages/core/src/service/transfer/tonService.ts]: Let's move this functions to new file - "multi Send Service" or similar?

File: packages/uikit/src/components/desktop/multi-send/MultiSendTable.tsx [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+                                    receiver: undefined,
+                                    amount: undefined,
+                                    comment: ''
+                                })
+                            }
+                        >
+                            Add More
+                        </Button>
+                        <Spacer />
+                        <MultiSendFooter
```
Comment (KuznetsovNikita) [inline packages/uikit/src/components/desktop/multi-send/MultiSendTable.tsx:204]: Screenshot 2024-04-29 at 2 57 24 PM

Could you please fix layout for buttons
thanks

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                        ) : (
+                            <>
+                                <Label2>≈ {tonFee}</Label2>
+                                <Body3>{fiatFee}</Body3>
+                            </>
+                        )}
+                    </FeeContainer>
+                </ListItemStyled>
+            </ListBlockStyled>
+            <ButtonBlock
```
Comment (KuznetsovNikita) [inline packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Screenshot 2024-04-29 at 3 00 22 PM

Could you please write an error with happen on this screen 
"The maximum number of transactions for Wallet Version 4 is four. If you need to send more transactions at one time, please use Wallet Version 5."

and "The maximum number of transactions for Wallet Version 5 is 255."

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+                form={formTokenized}
+                asset={asset}
+                feeEstimation={estimateData?.fee.weiAmount}
```
Comment (KuznetsovNikita) [inline packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: and let's replace with view to confirm ton connect view, where we have a row for every internal transaction

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+                onSuccess={() => {
+                    setTimeout(onClose, 2000);
```
Comment (KuznetsovNikita) [inline packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Could you please navigate to history screen after success multi send?

File: packages/core/src/service/transfer/jettonService.ts (hunk 1 above)
Comment (siandreev) [reply packages/core/src/service/transfer/jettonService.ts]: There well be a computation exception Out of gas. @mr-tron suggested using 10000 nanoton instead

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx (hunk 5 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Will do that in the Form Screen

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx (hunk 6 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: We already have a special designed modal for that, will implement

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx (hunk 7 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: 👌

Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Done

File: packages/uikit/src/components/desktop/multi-send/MultiSendTable.tsx (hunk 4 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendTable.tsx:204]: Done

File: packages/core/src/service/transfer/tonService.ts (hunk 3 above)
Comment (siandreev) [reply packages/core/src/service/transfer/tonService.ts]: Done

**Comment by KuznetsovNikita** [comment]
Screenshot 2024-04-30 at 11 41 42 AM

Button Continue is not disabled for non pro

**Comment by KuznetsovNikita** [comment]
Screenshot 2024-04-30 at 11 57 43 AM

Could you please show w5 for every one on this screen, as soon as you have a link from multi send page

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx (hunk 5 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Done

File: packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx (hunk 6 above)
Comment (siandreev) [reply packages/uikit/src/components/desktop/multi-send/MultiSendConfirmNotification.tsx]: Done

**Comment by siandreev** [comment]
> Screenshot 2024-04-30 at 11 57 43 AM
> Could you please show w5 for every one on this screen, as soon as you have a link from multi send page

Сейчас так и есть

**Pull request merged**